### PR TITLE
Switch to Jenkinsfile for jenkins job configuration

### PR DESCRIPTION
**Pull request by reaperhulk**
This is WIP until FreeBSD 11 is added (and it actually works)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    [
+        label: 'windows64',
+        toxenvs: ['py26', 'py27', 'py33', 'py34', 'py35', 'py36'],
+    ],
+]
+
+def build(label, toxenv) {
+    try {
+        checkout scm
+        if (label.startsWith("windows")) {
```
Comment (alex) [inline Jenkinsfile]: All of our builders appear to be windows for now?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    ],
+    [
+        label: 'windows64',
+        toxenvs: ['py26', 'py27', 'py33', 'py34', 'py35', 'py36'],
+    ],
+]
+
+def build(label, toxenv) {
+    try {
+        checkout scm
```
Comment (alex) [inline Jenkinsfile]: Should this be a separate stage?

File: Jenkinsfile (hunk 1 above)
Comment (reaperhulk) [inline Jenkinsfile]: (FreeBSD comes next to match the existing bcrypt PR builder)

File: Jenkinsfile (hunk 2 above)
Comment (reaperhulk) [inline Jenkinsfile]: Yeah probably so

**Comment by reaperhulk** [comment]
This is now ready for review. One caveat: the ansiColor build wrapper is currently of limited utility and doesn't work except in the combined console output view.

**Comment by reaperhulk** [comment]
Also, the separation of scm checkout into a different stage makes the non-blue ocean view pretty unreadable.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        currentBuild.result = 'FAILURE'
+        throw e
+    } finally {
+        deleteDir()
+    }
+
+}
+
+def builders = [:]
+for (x in configs) {
```
Comment (alex) [inline Jenkinsfile]: Real variable name.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    def label = x["label"]
+    def toxenvs = x["toxenvs"]
+
+    for (y in toxenvs) {
```
Comment (alex) [inline Jenkinsfile]: Real vairabla nae

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        timeout(time: 5, unit: 'MINUTES') {
+            if (label.startsWith("windows")) {
+                bat """
+                    @set PATH="C:\\Python27";"C:\\Python27\\Scripts";%PATH%
+                    tox -r -e $toxenv
+                """
+            } else {
+                ansiColor('xterm') {
+                    sh """
+                    tox -r -e $toxenv --  --color=yes
```
Comment (alex) [inline Jenkinsfile]: Can this just be a one liner?

**Comment by alex** [comment]
Besides those comments, is there anything blocking landing thsi?

**Pull request merged**